tests/on-change.test.js

```diff
@@ -751,3 +751,35 @@ test('should not wrap a proxied object in another proxy', t => {
 
 	t.is(proxy.b, proxy.c);
 });
+
+test('path should be the shorter one in the same object for circular references', t => {
+	const layer1 = {group: null, val: 0};
+	const layer2 = {group: null, val: 0};
+	const layer3 = {group: null, val: 0};
+	const group = {
+		layers: [layer1, layer2, layer3],
+		val: 0,
+	};
+	layer1.group = group;
+	layer2.group = group;
+	layer3.group = group;
+
+	let resultPath = null;
+
+	const proxy = onChange(group, path => {
+		resultPath = path;
+	});
+
+	proxy.layers[0].val = 11;
+	t.is(resultPath, 'layers.0.val');
+
+	proxy.layers[0].group.val = 22;
+	t.is(resultPath, 'layers.0.group.val');
+
+	proxy.layers[0].group.layers[0].val = 33;
+	t.is(resultPath, 'layers.0.val');
+
+	proxy.layers[1].group.layers[0].group.layers[1].group.layers[1].group.layers[2].val = 33;
+	t.is(resultPath, 'layers.2.val');
+});
+
```

Comment:
This test does not test the mentioned infinite loop. It would be good with a test that would have caused infinite loop before this pull request.

Reply:
Sorry, actually "infinite loop" was not the right term here.  
The more we were modifying the object property, the more the path was getting longer.
This mean that after the first call of callback we got `proxy.layers[1].group.layers[0].val`.
After the second call we got `proxy.layers[1].group.layers[0].group.layers[2].group.layers[0].val`.
After the third call we got `proxy.layers[2].group.layers[2].group.layers[1].group.layers[0].group.layers[2].group.layers[0].val`.
... and so on.

Example fo logs:
![image](https://github.com/sindresorhus/on-change/assets/15923531/5302c39b-92e7-48b4-b7fc-138a2dedb5f9)

And after 1 minute of usage of our app:
![image](https://github.com/sindresorhus/on-change/assets/15923531/9f5e10cc-dc55-4ce6-92af-772a1f102ec4)

And actually we reached a memory limit because we were getting path that has Megabytes of data.

So, we cannot say this is a infinite loop.  
But this tests actually exactly covers our problem.

Reply:
Ok. I would reword the comment then.

Reply:
Ok, I've rewrited the comment.